File: sdks/java/src/main/java/com/boltprediction/sdk/model/BatchReportResponse.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 批量报告响应 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class BatchReportResponse {

    @JsonProperty("total")
    private Integer total;

    @JsonProperty("success")
    private Integer success;

    @JsonProperty("failed")
    private Integer failed;

    @JsonProperty("results")
    private List<PeriodicReportResponse> results;

    @JsonProperty("errors")
    private Map<String, String> errors;

    public Integer getTotal() {
        return total;
    }

    public void setTotal(Integer total) {
        this.total = total;
    }

    public Integer getSuccess() {
        return success;
    }

    public void setSuccess(Integer success) {
        this.success = success;
    }

    public Integer getFailed() {
        return failed;
    }

    public void setFailed(Integer failed) {
        this.failed = failed;
    }

    public List<PeriodicReportResponse> getResults() {
        return results;
    }

    public void setResults(List<PeriodicReportResponse> results) {
        this.results = results;
    }

    public Map<String, String> getErrors() {
        return errors;
    }

    public void setErrors(Map<String, String> errors) {
        this.errors = errors;
    }

}

File: sdks/python/bolt_prediction_sdk/models/batch_report_response.py
"""
BatchReportResponse 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class BatchReportResponse(SDKBaseModel):
    """批量报告响应"""

    total: Optional[int] = Field(description="总数", default=0)
    success: Optional[int] = Field(description="成功数量", default=0)
    failed: Optional[int] = Field(description="失败数量", default=0)
    results: Optional[List[PeriodicReportResponse]] = Field(description="成功的报告列表", default=None)
    errors: Optional[Dict[str, str]] = Field(description="失败的节点及错误信息", default=None)
